Workaround for virtualenv package not working on python2 nix

Copied from https://github.com/NixOS/nixpkgs/issues/84568#issuecomment-632868701

diff --git a/src/virtualenv/discovery/py_info.py b/src/virtualenv/discovery/py_info.py
index 6f12128..98cf661 100644
--- a/src/virtualenv/discovery/py_info.py
+++ b/src/virtualenv/discovery/py_info.py
@@ -51,13 +51,17 @@ class PythonInfo(object):
         self.version = u(sys.version)
         self.os = u(os.name)

+        config_vars = None if sys.version_info.major is not 2 else sysconfig._CONFIG_VARS
+        base_prefix = None if config_vars is None else config_vars["prefix"]
+        base_exec_prefix = None if config_vars is None else config_vars["exec_prefix"]
+
         # information about the prefix - determines python home
         self.prefix = u(abs_path(getattr(sys, "prefix", None)))  # prefix we think
-        self.base_prefix = u(abs_path(getattr(sys, "base_prefix", None)))  # venv
+        self.base_prefix = u(abs_path(getattr(sys, "base_prefix", base_prefix)))  # venv
         self.real_prefix = u(abs_path(getattr(sys, "real_prefix", None)))  # old virtualenv

         # information about the exec prefix - dynamic stdlib modules
-        self.base_exec_prefix = u(abs_path(getattr(sys, "base_exec_prefix", None)))
+        self.base_exec_prefix = u(abs_path(getattr(sys, "base_exec_prefix", base_exec_prefix)))
         self.exec_prefix = u(abs_path(getattr(sys, "exec_prefix", None)))
